src/auth/token.ts

```diff
@@ -82,23 +82,23 @@ export class VaultTokenClient extends AbstractVaultClient {
             this.autoRenewErrorHandlers.add(onError);
         }
 
+        if (this.renewTimeout) {
+            clearTimeout(this.renewTimeout);
+        }
+
         const result = await this.renewSelf(undefined, true);
 
-        if (!this.autoRenewEnabled) {
-            setTimeout(() => {
-                this.autoRenew().catch((error) => {
-                    this.autoRenewErrorHandlers.forEach((handler) => {
-                        try {
-                            handler(error);
-                        } catch (handlerCallError) {
-                            // ignore errors from error handler
-                        }
-                    });
+        this.renewTimeout = setTimeout(() => {
```

Comment:
Wouldn't it have sufficed to call `this.renewSelf` (instead of calling `autoRenew` recursively) with `setInterval` instead of `setTimeout`?